src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/IOption2.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    internal interface IOption2
```

Comment:
I would have ideally like `IOption2` to be directly convertible to `IOption` by being its base type, but that cannot be done as `IOption` is public and `IOption2` has to be internal. This was the best I could do to ensure they have similar API surface.